server/src/main/java/io/crate/planner/optimizer/rule/MoveLimitBeneathUnion.java

```diff
@@ -64,27 +64,32 @@ public LogicalPlan apply(Limit limit,
                              NodeContext nodeCtx,
                              Function<LogicalPlan, LogicalPlan> resolvePlan) {
         Union union = captures.get(unionCapture);
+        var newLimit = getGlobalLimit(limit);
+        var newOffSet = Literal.of(0);
+
+        return new Limit(
+            new Union(
+                new Limit(union.lhs(), newLimit, newOffSet, false, false),
+                new Limit(union.rhs(), newLimit,newOffSet, false, false),
+                union.outputs()
+            ),
+            limit.limit(),
+            limit.offset(),
+            true,
+            limit.isPushedBeneathJoin()
+        );
+    }
+
+    static Literal<Integer> getGlobalLimit(Limit limit) {
```

Comment:
I think it's better to have this method in `io.crate.planner.operators.Limit` directly.